src/shared/copy.c

```diff
@@ -1124,6 +1130,16 @@ static int fd_copy_tree_generic(
                                          override_gid, copy_flags, denylist, hardlink_context, display_path,
                                          progress_path, progress_bytes, userdata);
 
+        if (hashmap_contains(denylist, st)) {
+                DenyType t = PTR_TO_INT(hashmap_get(denylist, st));
```

Comment:
as suggested below, if you add DENY_DONT with a value of zero, you could do the hashmap_get() line without the hashmap_contains() check first, and it would contain a pretty DENY_DONT whenever an inode is not listed.